### PR TITLE
docs: added type hints and comments to Datastore samples

### PR DESCRIPTION
**Pull request by craiglabenz**
Improved clarity around Datastore samples for beginners

### CODE REVIEW

File: datastore/cloud-client/tasks.py [hunk 1]
```diff
@@ -25,47 +26,60 @@ def create_client(project_id):
 # [END datastore_build_service]
 
 
+import google.cloud.datastore  # noqa
```
Comment (craiglabenz) [inline datastore/cloud-client/tasks.py:29]: Adding this import to allow for full type-hints

**Comment by craiglabenz** [comment]
This PR is down to a linting warning that is purely an artifact of how we're dividing up code so they appear nicely in rendered snippets on devsite. I think it is good to go, otherwise.

File: datastore/cloud-client/snippets.py [hunk 2]
```diff
@@ -110,10 +229,13 @@ def properties(client):
...
 
 def array_value(client):
-    key = client.key("Task")
     # [START datastore_array_value]
+    key = client.key("Task")
     task = datastore.Entity(key)
     task.update({"tags": ["fun", "programming"], "collaborators": ["alice", "bob"]})
+
+    # To access array values:
+    task["tags"]  # ["fun", "programming"]
```
Comment (crwilcox) [inline datastore/cloud-client/snippets.py:238]: this ticks out as a bit strange. Do we have a read array sample? if not including here may be fine. What do other langs do? Should they also be updated?

Comment (craiglabenz) [reply datastore/cloud-client/snippets.py:238]: Resolving this with additional tickets, per our conversation.

File: datastore/cloud-client/snippets.py [hunk 3]
```diff
@@ -16,8 +16,128 @@
 import datetime
 from pprint import pprint
 
-from google.cloud import datastore
 import google.cloud.exceptions
+from google.cloud import datastore
```
Comment (engelke) [inline datastore/cloud-client/snippets.py:20]: You could add "# noqa: I100" to the end of this line (and perhaps the line before it) to suppress the lint message and declare that the order is intentional, not an error.

File: datastore/cloud-client/snippets.py [hunk 4]
```diff
@@ -66,6 +186,9 @@ def basic_entity(client):
             "description": "Learn Cloud Datastore",
         }
     )
+    # Access fields
+    task["category"]  # "Personal"
+    task["priority"]  # 4
```
Comment (engelke) [inline datastore/cloud-client/snippets.py:191]: I find this confusing. I get showing how to access a field, but statements that are just a value aren't something I'd expect. You might want to consider doing something with these values, such as logging them, to show not only how to reference them but also how to use the result.

**Review by engelke — APPROVED** [review]
LGTM, though I recommend suppressing the import order lint issue with # noqa.

**Review by crwilcox — APPROVED** [review]

**Pull request merged**